### DISCUSSION TITLE
Transmission UI/UX redesign - Gnome 3 concept

### DISCUSSION THREAD

**Discussion by budislav** [transmission/transmission#1943] [Ideas]
I made a concept for Transmission UI to go further with the latest Gnome Adwaita look. <3
I have done it with Pencil app Gnome 3 widgets and Inkscape for some icons.
![transmissiongtk3_new](https://user-images.githubusercontent.com/3619927/34331912-68627a3a-e92b-11e7-847a-4b65b17b412d.png)
![transmissiongtk3_drop1](https://user-images.githubusercontent.com/3619927/34330952-9c748844-e922-11e7-805c-9e006ef0a328.png)
![transmissiongtk3_drop2](https://user-images.githubusercontent.com/3619927/34330961-b1855ce0-e922-11e7-8386-badbd46a59b4.png)
![transmissiongtk3_drop3](https://user-images.githubusercontent.com/3619927/34330967-c25bf52e-e922-11e7-8965-8367bb12ed01.png)
![transmissiongtk3_drop4](https://user-images.githubusercontent.com/3619927/34330976-db18f922-e922-11e7-9514-90a108a66508.png)
![transmissiongtk3_new_search](https://user-images.githubusercontent.com/3619927/34330983-ed5b4bee-e922-11e7-9789-1414efd1fe26.png)
Some of my previous works
https://budislavtvp.deviantart.com/art/Zyn-UI-UX-Design-Concept-Fusion-455890191
https://budislavtvp.deviantart.com/art/LMMS-UI-UX-Design-Concept-Single-Window-523696539

**Comment by felselva** [comment]
I'm working on a GUI that uses transmission-daemon and transmission-remote.

![](https://pbs.twimg.com/media/DSFeI8dX0AAeSSi.png)
https://twitter.com/ferreiradaselva/status/946141170475896832

I have a repo for it, too (but it's a bit outdated, I will update soon): https://github.com/ferreiradaselva/transmitter

It's an early stage, but I have already some communication between the GUI and transmission-daemon via transmission-remote. In the screenshot, for example, the name of the torrent, the upload/download values are extracted using the pipe output from transmission-remote.

**Comment by mikedld** [comment]
@derekstavis made some progress on GTK+ client redesign a while back with initial design work done by @aldomann (see the link at derekstavis/transmission#3). I've rebased his work to a more recent Transmission version in wip/gtk-3.14 branch if you want to try it out.

**Comment by budislav** [comment]
@mikedld thanks for answer.
That concept is great, I didn't knew for it :)

Anyway I really like controls in limit speed menu and that green bar looks awesome when torrent seeding. I like tabs in search/filter tab but I think those tabs should be always visible in the middle of the header bar instead of dropdown lists on mine mockup. Also I like the way you can see torrent options on the left side of each torrent. Also upload and download status on the bottom in the middle it is better place then on right side.

I don't like one start and pause buttons on the top like open, add and remove buttons. I think it works better if you can start and pause each torrent separately, like downloads in browser. Maybe that button for removing torrent on my mockup should be visible only on select or hover. Also I thought it would be faster if you can sort torrents with separate button for that, it's just one step faster because it is more frequently used option? Also I think those buttons for speed limit are better to be in header bar, they not belong to status bar?

I think it would be cool if we have some message on the bottom in the status bar that reflect upload status. So if you upload the whole torrent message would be "You are awesome!" or something, if you turned off upload to say "You are not fair" and sad emoticon or something else :) When you seeding to see "Keep going".. Some kind of pleasant short message that would be funny but also to motivate you to keep seeds.

Anyway I would be happy to see anything from these concepts in Transmission :)

**Comment by RealKindOne** [comment]
How about an option to keep the current UI? I'm not a fan of those pictures.

**Comment by AaronBottegal** [comment]
Looks terrible. Leave the window management to the window manager layer. Don't integrate terrible custom UI features that make all your apps impossible to navigate unless you learn their terrible designs. This is a step back when actually using the program. There is no sorting, there is less room for features in menus, and it really is just a terrible idea.

**Comment by corebots** [comment]
would love to see the compact headerbar finally implemented and shipped! good work on the designs, thanks!

**Comment by opportus** [comment]
Hi @mikedld, do you plan merging `wip/gtk-3.14` sooner or later? How people can help about this issue?

**Comment by JaneSmith** [comment]
Any update on this? The concept looks fantastic, and it's now been over two years. I switched to Fragments as my BitTorrent client because of the UI.

**Comment by vassilisw** [comment]
I hope the developers keep the layout of the window as it is. A layout that real user interface scientists designed, specialists that know white space is _essential_ for the user to _quickly_ spot an item, or _distinguish_ elements and their _role_. Trends to change are mostly adopted by commercial companies like Apple or Microsoft just to keep their clients desire of "going forward" and keep their interest alive and gain user attention. But efficiency? No! Why?...  

- You can drag n drop windows from header's buttons!! As inconvenient as it gets!
- It is often compared with macOS but this is completely wrong because macOS has and always will have menu bar.
- lucking the menu bar, users have to seek for functionality among buttons placed left or right **without any standard design**.
- Buttons are have to be left-clicked -> released -> re-click instead just hold-click -> release-click on the item!!
- Keyboard shortcuts? Without the menu bar (compined with the alt button), there is no way to reach an item with just the keyboard!

I don't think that I should mention more and more CSD inconveniences, although I could... easily.  

Transmission team: Good work!

**Comment by nekohayo** [comment]
> Buttons are have to be left-clicked -> released -> re-click instead just hold-click -> release-click on the item!!

Buttons are meant to be clicked to be activated, not dragged around kicking and screaming until you release them into a river.

> Keyboard shortcuts? Without the menu bar (compined with the alt button), there is no way to reach an item with just the keyboard!

Yes there is. App developers should bind the F10 keyboard shortcut to the main menubutton, like other apps do. It's pretty much a de-facto standard among CSD apps that have a menubutton.

**Comment by vassilisw** [comment]
> Buttons are meant to be clicked to be activated, not dragged around kicking and screaming until you release them into a river.

I said buttons because buttons are used. The user needs to move the window somehow! (please don't mention alt + mouse).

> Yes there is. App developers should bind the F10 keyboard shortcut to the main menubutton, like other apps do. It's pretty much a de-facto standard among CSD apps that have a menubutton.

But menu-button is not the only button in the header is it? What about the other ones? What about "What is this or that button hides?", as I said.. no standard design!

Aha, so developers do not forget to bind F10 key like Simple Scan devels did! (just tried it).

And what about applications with tenths of menu items like GIMP that it is suggested by Gnome to be ported! I can't wait to see this!

For all of you that do not hate the hater :) you can take a look at [this](https://pointieststick.com/2019/03/06/more-on-headerbars-rebuttals/).

I was expecting thumps down but guys, I do not comment to argue but to make a point. Whoever is catching it is fine!  

Thanks for reading.

**Comment by nekohayo** [comment]
> I said buttons because buttons are used. The user needs to move the window somehow! (please don't mention alt + mouse).

I was responding to that specific sentence of yours in your comment. The previous sentence:

> You can drag n drop windows from header's buttons!! As inconvenient as it gets!

...contradicts your later sentence I was quoting, within the same comment. On one hand you're complaining that you can't click-drag-release to activate a button (hence my initial reply that this is not how a button is supposed to work) and on the other hand you complain that you... _can_ drag a headerbar by the buttons to move the window? So which is it? It looks like you're really trying to invent problems here.

> Aha, so developers do not forget to bind F10 key like Simple Scan devels did! (just tried it).

It works for me here (I just tried it).
As for individual apps that don't implement it yet, if some devs forgot, file a bug report on those apps.

> And what about applications with tenths of menu items like GIMP that it is suggested by Gnome to be ported! I can't wait to see this!

Nobody is talking about GIMP here, and nobody reasonable anywhere else is asking hyper-complicated applications like GIMP, LibreOffice and Inkscape to replace menubars, those are pretty much expected to retain their menubar-and-multi-toolbar designs forever. However, simple applications, like Transmission, can definitely achieve this without much difficulty.


The rest of your comment is just more strawman arguments and trolling, so I'm not responding to that. Be constructive instead of trying to stop people from improving software.

**Comment by vassilisw** [comment]
You have misunderstood the drag thing. I do not have the need to drag things around without a reason.
So, which one is it...
- Buttons are kind of replacing the menu items. Menu items have the click-drag(move mouse over an item)-release(to select the item) concept and buttons don't. So minus in convenience and functionality. "Problem invented!"
- The user should have a way to move the window around. When you tell the user click on the button to move the window is also inconvenient. "Problem invented!"
- Well Simple Scan dont work for me BUT if the menu is used, no developer nor user ("file a bug!") has to deal with things that are already covered. "Problem invented!"
- Simple CSD applications among classic ones in the same desktop break the desktop consistency. I quote from [XApps](https://linuxmint-developer-guide.readthedocs.io/en/latest/xapps.html) 

> The idea behind this project is to replace applications which no longer integrate properly outside of a particular environment (this is the case for a growing number of GNOME applications) and to give our desktop environments the same set of core applications, so that each change, each new feature being developed, each little improvement made in one of them will benefit not just one environment, but all of them.

Problem invented?

Really, I am sorry If I have offended you with my comments. It is a nice effort you are making to improve the software but is just your way of improving it. It is best to have an option in user preferences, and let the user decide for which ui to use (or just fork). Maybe this way the transmission team will show some interest. Otherwise I really doubt if they ever adopt CSD. Hence this 3 year-old thread.

Try to **listen** to other opinions.

**Comment by xlucn** [comment]
Maybe it's easier to ask other programmers to make a front end with this design for transmission-daemon.

**Comment by budislav** [comment]
This is so old, 4 years now :D If I am doing this right now, it would look different.

Fragments is developed with Gnome on mind.

![fragments@1x](https://user-images.githubusercontent.com/3619927/137603937-095feaf8-962e-4551-acd8-6f0a8285f195.png)

https://tobiasbernard.com/assets/fragments@1x.png

I would definitely go with adaptive design, more empty space and other ux issues I see now :D

**Comment by ckerr** [comment]
I should have linked to this sooner, but see also: https://github.com/transmission/transmission/issues/1686

The mockups here and in that link have similar strengths and weaknesses. I think both look very good. And in general I'd love it if someone wants to modernize transmission-gtk to make it its look-and-feel more like a modern Gnome app.

Anyone who wants to do some work on this, please see [this writeup](https://github.com/transmission/transmission/issues/1686#issuecomment-846044730) I made for the mockup in 1686. In particular, Transmission tries to be an app that has a simple & easy API _that also_ lends itself to power users seeding thousands of torrents. All of the redesign mockups I've seen seem to believe that people only have a few torrents at a time. Redesigns need to be easy out-of-the-box _but also_ scale well for hardcore users.

None of that is to take away from these proposals. If someone feels strongly about this and wants to mockups and dig into the code, I'd _love_ to give feedback on the mockups and code review on any PRs that come out of it.

**Comment by comradekingu** [comment]
@budislav 
I think it should be
Queue
Downloading
Seeding

That way torrents can move down the list.
If you can make it not focus-shift by moving the category names or preferably using icons ⏸ ▶ ⇑ (or something) that would be even better.
